GitUI/BuildServerIntegration/BuildServerWatcher.cs

```diff
@@ -79,11 +79,23 @@ public async Task LaunchBuildServerInfoFetchOperationAsync()
 
             var fullDayObservable = buildServerAdapter.GetFinishedBuildsSince(scheduler, DateTime.Today - TimeSpan.FromDays(3));
             var fullObservable = buildServerAdapter.GetFinishedBuildsSince(scheduler);
-            var fromNowObservable = buildServerAdapter.GetFinishedBuildsSince(scheduler, DateTime.Now);
 
             bool anyBuilds = false;
             var delayObservable = Observable.Defer(() => Observable.Empty<BuildInfo>()
                                                                    .DelaySubscription(anyBuilds ? ShortPollInterval : LongPollInterval));
+
+            var shouldLookForFinishedBuilds = false;
+            DateTime nowFrozen = DateTime.Now;
+            var fromNowObservable = Observable.Defer(() =>
+            {
+                // All finished builds have already been retrieved,
+                // so looking for new finished builds make sense only if running builds have been found previously
+                return !shouldLookForFinishedBuilds
+                    ? Observable.Empty<BuildInfo>()
+                    : buildServerAdapter.GetFinishedBuildsSince(scheduler, nowFrozen)
```

Comment:
consider using `Observable.If()`, based on [defer source code](https://github.com/dotnet/reactive/blob/f650dded4d6d6ced6178f864f2eaecdbb57d5b43/Rx.NET/Source/src/System.Reactive/Linq/Observable/Defer.cs#L21) and [if source code](https://github.com/dotnet/reactive/blob/3f2c1f02018c07c3f34f9d54c844b358ad671721/Rx.NET/Source/src/System.Reactive/Linq/Observable/If.cs#L21) `if(condition, observable)` should be equivalent to  `defer(condition ? observable : empty)`

Reply:
Thanks, I will have a look. I don't know if that's compatible with the `.Finally()` but I will try.